### PR TITLE
client/batch: allow tokenCh of batch controller to be nil

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: ref #8690.

### What is changed and how does it work?

```commit-message
Allow `tokenCh` of batch controller be nil.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: client/pkg/batch/batch_controller.go [hunk 1]
```diff
@@ -88,17 +91,19 @@ func (bc *Controller[T]) FetchPendingRequests(ctx context.Context, requestCh <-c
...
-			return ctx.Err()
-		case req := <-requestCh:
-			// Start to batch when the first request arrives.
-			bc.pushRequest(req)
-			// A request arrives but the token is not ready yet. Continue waiting, and also allowing collecting the next
-			// request if it arrives.
-			continue
-		case <-tokenCh:
-			tokenAcquired = true
+		if tokenCh != nil {
```
Comment (bufferflies) [inline client/pkg/batch/batch_controller.go:94]: when the token channel is nil in product branch? I see only test maybe use nil.

Comment (JmPotato) [reply client/pkg/batch/batch_controller.go:94]: I will use it in a later PR.

**Review by MyonKeminta — COMMENTED** [review]
Rest LGTM

**Review by MyonKeminta — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

File: client/pkg/batch/batch_controller.go [hunk 2]
```diff
@@ -60,13 +60,18 @@ func NewController[T any](maxBatchSize int, finisher FinisherFunc[T], bestBatchO
 // It returns nil error if everything goes well, otherwise a non-nil error which means we should stop the service.
 // It's guaranteed that if this function failed after collecting some requests, then these requests will be cancelled
 // when the function returns, so the caller don't need to clear them manually.
+// `tokenCh` is an optional parameter:
```
Comment (rleungx) [inline client/pkg/batch/batch_controller.go:63]: Shall we use something like
```
FetchPendingRequests(ctx context.Context, requestCh <-chan T, maxBatchWaitInterval time.Duration, tokenCh ...chan struct{}) (errRet error) {
```

Comment (JmPotato) [reply client/pkg/batch/batch_controller.go:63]: It looks good to me, but I prefer keeping the current way to force the caller to decide whether `tokenCh` is needed by explicitly passing it in.

**Review by rleungx — APPROVED** [review]

**Pull request merged**